tt_metal/llrt/hal.cpp

```diff
@@ -2,45 +2,55 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#include <cstdlib>
+
 #include "hal.hpp"
-#include "tt_metal/third_party/umd/device/tt_soc_descriptor.h"
-#include "tt_metal/third_party/umd/device/tt_arch_types.h"
+
+#include "tt_metal/common/tt_backend_api_types.hpp"
+#include "tt_metal/common/assert.hpp"
+#include "tt_metal/third_party/umd/device/cluster.h"
 
 namespace tt {
 
 namespace tt_metal {
 
-Hal hal;
-
-// This back poitner is a little clunky but necessary at least for now
-Hal::Hal() : initialized_(false) {
-}
-
-void Hal::initialize(tt::ARCH arch) {
-
-    const std::lock_guard<std::mutex> lock(this->lock);
-
-    if (!this->initialized_) {
-        switch (arch) {
-        case tt::ARCH::GRAYSKULL:
-            initialize_gs();
-            break;
-
-        case tt::ARCH::WORMHOLE_B0:
-            initialize_wh();
-            break;
-
-        case tt::ARCH::BLACKHOLE:
-            initialize_bh();
-            break;
-
-        default:
-            TT_THROW("Unsupported arch for HAL");
+// Hal Constructor determines the platform architecture by using UMD
+// Once it knows the architecture it can self initialize architecture specific memory maps
+Hal::Hal() {
+    this->arch_ = tt::ARCH::Invalid;
+    if(std::getenv("TT_METAL_SIMULATOR_EN")) {
```

Comment:
this looks like it was copied from `Cluster::detect_arch_and_target()` could you just:
```suggestion
    this->arch_ = tt::Cluster::instance().arch()
```
since Cluster is a singleton

Reply:
Latest commit uses common inline function.